source/core/MicroBitFiber.cpp

```diff
@@ -754,6 +828,8 @@ void verify_stack_size(Fiber *f)
 
         // Recalculate where the top of the stack is and we're done.
         f->stack_top = f->stack_bottom + bufferSize;
+
+        currentFiber = prevCurrFiber;
     }
```

Comment:
is this just for the case where the code may be pre-empted by an interrupt that scans the fiber list? if so, it's a short piece of code, can we protect with a more traditional disable/enable IRQ?